docs: add release notes for 8.12.0 (#5752)
canonical/snapcraft: docs/release-notes/snapcraft-8-12.rst

Snapcraft 8.12 brings the following features, integrations, and improvements.

New metadata linter
~~~~~~~~~~~~~~~~~~~

A :ref:`new linter <how-to-use-the-metadata-linter>` has been added to help identify
problems in the metadata of a snap project. For projects with ``grade: stable``, a
linter warning will be emitted if the ``title``, ``contact``, or ``license`` keys are
missing or malformed. In addition, an informational message will be emitted if the
``donation``, ``issues``, ``source-code``, or ``website`` keys are missing or

> malformed.

Author: How about "the linter raises a warning"?

> A :ref:`new linter <how-to-use-the-metadata-linter>` has been added to help identify
> problems in the metadata of a snap project. For projects with ``grade: stable``, a
> linter warning will be emitted if the ``title``, ``contact``, or ``license`` keys are
> missing or malformed. In addition, an informational message will be emitted if the
> ``donation``, ``issues``, ``source-code``, or ``website`` keys are missing or
> malformed.

Reviewer: We could go with "warn" and "log". But I'm worried that some users might consider logs to be messages in files, not messages on the CLI.

```suggestion
A :ref:`new linter <how-to-use-the-metadata-linter>` is available to help identify
problems in a project's metadata keys. If your project has a stable grade, the linter
warns you if the ``title``, ``contact``, or ``license`` keys are missing or
malformed. Likewise, if the ``donation``, ``issues``, ``source-code``, or ``website``
keys are missing or malformed, the linter logs it.
```

> malformed.

Author: Hmm. I think your proposed commit is the best I can come up with, I say let's roll with it. Thanks.
